CS_Ed_Week/CS_Ed_Week.md

```diff
@@ -77,6 +77,122 @@ A workshop for getting young people interested in computer science
+    - In pseudocode:
+      1. If `<condition>` is false, end.
+      2. Run `<do_thing>`.
+      3. Go to (1).
+    - If you are familiar with other languages, Python doesn't have a "do while" loop.
+  - `for <var-name> in <sequence>: <do_thing>` runs `<do_thing>` for each item in `<sequence>`. In each iteration of the loop, the value of `<var-name>` is set to the current item.
+    - ```python
+      my_things = ["cats", "dogs", "programming"]
+      for thing in my_things:
+          print("I like " + thing)
+      # prints:
+      # I like cats
+      # I like dogs
+      # I like programming
+      ```
+    - Note that the `<sequence>` is not modified.
+    - In pseudocode:
+      1. If there are no more items in `<sequence>`, end.
+      2. Get the next item of `<sequence>` and set the value of `<var-name>` to it.
+      3. Run `<do_thing>`.
+      4. Go to (1).
+    - If you are familiar with other languages, you'll notice this isn't a regular "for" loop, but rather a "for each" loop. Python doesn't have regular "for" loops.
+- Sometimes, in the middle of running a loop, you want to skip the rest of the current iteration or the whole loop altogether. This is usually because the current item of a sequence is not useful to you, or you've already found or calculated the value you wanted, or something else.
+  - `continue` skips the current iteration of the loop, immediately moving back to the top.
+    - ```python
+      my_things = ["cats", "dogs", "programming", "coding", "random stuff"]
+      for thing in my_things:
+          if len(thing) > 10:
+              continue # if it's more than 10 characters long, we don't like it!
+          print("I like " + thing)
+      # prints:
+      # I like cats
+      # I like dogs
+      # I like coding
+      ```
+  - `break` breaks out of the loop entirely, moving on to the next thing.
+    - ```python
+      has_dogs = False
+      my_things = ["cats", "dogs", "programming", "coding", "random stuff"]
+      for thing in my_things:
+          print("Checking " + thing + "...")
+          if thing == "dogs":
+              has_dogs = True
+              break # we already know the list has dogs, we don't need to check the rest!
+      # prints:
+      # "Checking cats..."
+      # "Checking dogs..."
+
+      if has_dogs:
+          print("The list has dogs!") # prints
+      else:
+          print("The list does not have dogs!") # does not print
+      ```
+- If you want to iterate through a range of numbers, you can use the `range` function.
+  - `range(stop)` iterates through the integers `0` (inclusive) through `stop` (exclusive):
+    ```python
+    for x in range(5):
+        print(str(x), end=" ") # space instead of newline at end
+    # prints 0 1 2 3 4
+    ```
+    ```python
+    num_list = [3, 5, 5, 100, -9, 0]
+    # compares adjacent numbers
+    # i is the left index of each pair
+    # the last index in the list isn't the left of any pair
+    for i in range(len(num_list)-1):
```

Comment:
Looks good, though I would suggest to make a more clear definition of `range()`. Other than that, it's good. 

"The range() function returns a sequence of numbers, starting from 0 by default, and increments by 1 (by default), and stops before a specified number." (W3School)

Reply:
imo the official definition is more confusing to someone inexperienced, but I think changing to "`range` returns a sequence of numbers..." is nice